### ISSUE TITLE
Add a simple one line measuring tool

### ISSUE THREAD

**Issue by courtneypattison** [Pattern-Projector/pattern-projector#70]
I had to make a few adjustments and cut a few simple rectangles today. It would have been nice to have a simple measuring tool that measures from one point to another for figuring out the sizes of the pieces. For rectangles, I prefer just using my ruler rather than the pattern piece for a straighter edge.

Inspired by [Adobe PDF Reader](https://helpx.adobe.com/ca/acrobat/using/grids-guides-measurements-pdfs.html)

**Comment by hluedeke** [comment]
Are you thinking of drawing your own rectangles with a predetermined size? Or are you saying you want to click on one location in the canvas then click on another location then have the distance be calculated? If the second, would you want snapping?

**Comment by courtneypattison** [comment]
The latter, snapping optional. I usually just want a rough estimate. What do you think?

**Comment by hluedeke** [comment]
Sure 🤷‍♀️ works for me. Maybe a ruler icon button that toggles the mode on/off. When mode is on, show crosshair cursor, one click starts the line, second click does the measure. Third click removes the previous measurement and starts a new one (?). Right-click cancels a measurement (?). IconButton or even hotkey (m key maybe?) would toggle the mode.

**Comment by courtneypattison** [comment]
That’s exactly what I was thinking!

**Comment by courtneypattison** [comment]
It could show the current measurement while moving the cursor after the first click.

**Comment by hluedeke** [comment]
That works too! I think we already have the orthogonal logic built in with that geometry library. So this feels pretty straightforward to implement.

**Comment by courtneypattison** [comment]
Yeah, I think so too. There are always little gotchas with the pdf versus css pixels though!

**Comment by roel-v** [comment]
As I was testing two days ago, I used Foxit pdf reader a bunch of times to double check measurements too, having this build into PP would be great (what I tested was measuring on the projected image with my measuring tape, then checking against measuring that same distance in the pdf viewer; I was off by maximum 2 mm, and 90% of the time 1mm or less. I feel that's more accurate than tracing on paper, cutting and then tracing again on the fabric, but proving that feeling would require some more structured testing). Anyway, one thing that I found/find useful in Foxit is the ability to constrain the angle at which measuring is happening by holding down the shift key. This restricts measuring to horizontal, vertical or at a 45 degree angle. When moving, a line is shown between the location the user clicked first and where the cursor is currently, so that you see what distance you're measuring. Holding shift basically snaps that line to the nearest of those 3.
